### ISSUE TITLE
How to compile FastBuild itself (fbuild/fbuildworker) on linux?

### ISSUE THREAD

**Issue by t31k3** [fastbuild/fastbuild#861]
It is almost impossible to find this info on google, as everything points to using FastBuild to compile and not Compiling Fast Build Itself, even less on linux.

I need this because Unreal Engine patches many source files of FastBuild v0.99 and then I need to compile fbuild and fbuildworker apps.

But I cant find any make file anywhere on the source code (the downloaded zip), neither any compilation instructions, not even on the site! (I didnt search enough?)

Any clues?

PS.: Ubuntu 20.04 here

**Comment by t31k3** [comment]
I found some things, fbuild builts itself and has some dependencies, still preparing it.

**Comment by ffulin** [comment]
Apologies for the lack of documentation on this.

FASTBuild is used to build itself as you noted. Generally, one version prior and any version in the future can be used. For example, to compile v0.99 should be possible with version v0.98 or later.

If you download the source archive from fastbuild.org, that also contains binaries for that version which can be used to compile that version.

**Comment by t31k3** [comment]
Thx, I liked the surprise! xD

**Comment by christophe-calmejane** [comment]
Hi,

I'd like to contribute to the project, but I cannot even get started on how to recompile FBuild on windows.
Cloned, switched to dev branch, tried to run "fbuild" inside the Code subfolder, but I got errors about clang-cl not being found.

Is there some documentation somewhere to get people started on compiling this projet, like required dependencies, env setup, ...?

I'm almost done writing a Fastbuild generator for CMake but now I need to check a few things with the latest FBuild version (and possibly improve it) but I really need to compile it :)

**Comment by claimred** [comment]
Some relevant threads about building FASTBuild https://github.com/fastbuild/fastbuild/issues/684#issuecomment-576507362, https://github.com/fastbuild/fastbuild/issues/153

When I need to compile it locally myself I look into CI configuration, actions, and replicate steps myself.

https://github.com/fastbuild/fastbuild/actions/runs/1072911304/workflow#L37
https://github.com/fastbuild/fastbuild/runs/3098203497?check_suite_focus=true

Basically one needs to configure paths or vendorize tools, set toolchain in fbuild.bff and fbuild.exe.

**Comment by christophe-calmejane** [comment]
Thanks a lot, I was searching github issues for "compile" keyword... not "build" ;)
